core/matching/side.go

```diff
@@ -409,6 +416,12 @@ func (s *OrderBookSide) fakeUncross(agg *types.Order, checkWashTrades bool) ([]*
 			checkPrice = func(levelPrice *num.Uint) bool { return levelPrice.GTE(agg.Price) }
 		}
 
+		// first check for volume between the theoretical best price and the first price level
+		_, oo := s.uncrossOffbook(len(s.levels)-1, fake, idealPrice, true)
```

Comment:
these functions calls do not commit any state to the pools right?

Reply:
No state change, just returns some orders. The state will only change later down the line when we're in `handleConfirmation()` and the position engine gets updated when the trades are finalised.